controllers/bsl.go

```diff
@@ -215,3 +221,23 @@ func (r *VeleroReconciler) validateProviderPluginAndSecret(bslSpec velerov1.Back
 	}
 	return nil
 }
+
+func (r *VeleroReconciler) ensureBSLProviderMapping(velero *oadpv1alpha1.Velero) error {
+
+	providerBSLMap := map[string]int{}
+	for _, bsl := range velero.Spec.BackupStorageLocations {
+		provider := bsl.Provider
+
+		if providerBSLMap[provider] == 0 {
```

Comment:
Why do you need this? Can't you just always run `providerBSLMap[provider]++`?

Reply:
Done